service/worker/scheduler/workflow.go

```diff
@@ -1180,8 +1180,10 @@ func (s *scheduler) startWorkflow(
 	nominalTimeSec := start.NominalTime.AsTime().UTC().Truncate(time.Second)
 	workflowID := newWorkflow.WorkflowId
 	if start.OverlapPolicy == enumspb.SCHEDULE_OVERLAP_POLICY_ALLOW_ALL || s.tweakables.AlwaysAppendTimestamp {
-		// must match AppendedTimestampForValidation
-		workflowID += "-" + nominalTimeSec.Format(time.RFC3339)
+		if !s.StartScheduleArgs.Schedule.Policies.GetKeepOriginalWorkflowId() {
```

Comment:
I am not sure what should take precedence: clients request not to append timestamp or AlwaysAppendTimestamp?
@dnr, @yiminc

Reply:
"AlwaysAppendTimestamp" should now mean "append timestamp unless requested not to". i.e. client request. Except ALLOW_ALL requires the timestamp always, so the logic needs to be turned around. Also we should validate that in CreateSchedule/UpdateSchedule

Reply:
Ah we allow empty workflow ID. I missed that, thanks.